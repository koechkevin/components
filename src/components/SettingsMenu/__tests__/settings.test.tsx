import { Link, MemoryRouter } from 'dva/router';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { SettingsMenu } from '../../..';

describe('Settings Menu Test', () => {
  it('mounts without crashing', () => {
    const wrapper = mount(<SettingsMenu routes={[]} location={{ pathname: '' }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders all routes appropriately', () => {
    const routes = [
      {
        name: 'Account',
        path: '/app/settings/account',
        routes: [
          {
            name: 'Profile',
            path: '/app/settings/account/profile',
          },
          {
            name: 'Account',
            path: '/app/settings/account/account',
          },
        ],
      },
    ];
    const wrapper = mount(
      <MemoryRouter>
        <SettingsMenu companyName="Aurora Unit Test" routes={routes} location={{ pathname: '/app/settings/account/profile' }} />
      </MemoryRouter>,
    );
    const link = wrapper.find(Link);
    expect(link).toHaveLength(2);
  });
});
