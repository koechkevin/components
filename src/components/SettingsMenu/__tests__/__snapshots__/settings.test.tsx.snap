// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Menu Test mounts without crashing 1`] = `
<SettingsMenu
  location={
    Object {
      "pathname": "",
    }
  }
  routes={Array []}
>
  <div
    className="settingsSidebar"
  >
    <div
      className="items"
    />
  </div>
</SettingsMenu>
`;
